reflect.go

```diff
@@ -334,8 +327,35 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 	panic("unsupported type " + t.String())
 }
 
-// Refects a struct to a JSON Schema type.
+func (r *Reflector) reflectCustomType(definitions Definitions, t reflect.Type) *Type {
+	if t.Kind() == reflect.Ptr {
+		return r.reflectCustomType(definitions, t.Elem())
+	}
+
+	if t.Implements(customType) {
+		v := reflect.New(t)
+		o := v.Interface().(customSchemaType)
+		st := o.JSONSchemaType()
+		definitions[r.typeName(t)] = st
+		if r.DoNotReference {
+			return st
+		} else {
+			return &Type{
+				Version: Version,
+				Ref:     "#/definitions/" + r.typeName(t),
+			}
+		}
+	}
+
+	return nil
+}
+
+// Reflects a struct to a JSON Schema type.
 func (r *Reflector) reflectStruct(definitions Definitions, t reflect.Type) *Type {
+	if st := r.reflectCustomType(definitions, t); st != nil {
```

Comment:
This is also why I moved this call up, before checking for `r.IgnoredTypes`, so that any recursive call to `reflectStruct` behaves the same as call to `reflectTypeToSchema` (that is, `IgnoredTypes` is ignored if struct implements `JSONSchemaType`).